### PR TITLE
Fixes #23409 - Exit properly on failed hammer ping

### PR DESCRIPTION
**Pull request by johnpmitsch**
during service restart

### CODE REVIEW

**Comment by johnpmitsch** [comment]
Why did we originally rescue StandardError here?

File: definitions/procedures/service/restart.rb [hunk 1]
```diff
@@ -28,11 +28,11 @@ def hammer_ping_retry
             break
           elsif retry_count < (RETRIES_FOR_SERVICES_RESTART - 1)
             apply_sleep_before_retry(spinner, result)
+          else
+            raise "Hammer ping failed!"
```
Comment (stbenjam) [inline definitions/procedures/service/restart.rb]: Travis test failing because not using single quotes

**Comment by iNecas** [comment]
@johnpmitsch looking at `git blame`, you tell me :)

**Comment by johnpmitsch** [comment]
@iNecas hah :smile: I copied this from another area when I added the service command, I can hunt it down in an old commit to find the original author

**Comment by kgaikwad** [comment]
@johnpmitsch,

IIRC when I have added hammer ping retry changes, firstly I did it using `Timeout`. So for `Timeout` I have added this rescue block `StandardError`. During PR review I have removed `Timeout` but rescue block remains as it is.

**Comment by iNecas** [comment]
Deferring to @ntkathole for testing. ACK codewise

**Comment by iNecas** [comment]
Thanks @johnpmitsch and @ntkathole

**Pull request merged**